src/util/decision_tree/sample.json

```diff
@@ -0,0 +1,99 @@
+{ "collective": "reduce",
+    "children":
+    [
+        {
+            "type": "comm_size", "comparison": "lt", "value": 32,
+            "children":
+            [
+                {
+                    "type": "node_size", "comparison": "lte", "value": 8, "algorithm":
+                    {
+                        "mpir": "device",
+                        "device":
+                        {
+                            "algorithm": "alpha",
+                            "netmod":
+                            {
+                                "algorithm": "binomial", "kval": 4
```

Comment:
This is the reason we have introduced containers.
coll_id/name -> binomial, it may have parameters like radix (4 here)

So almost each json leaf is transalted into container

Reply:
Yeah, they act very similar. Containers = more parsing at the init time. Strings = more parsing at calling time.